src/mlpack/core/optimizers/adam/optimisticadam_update.hpp

```diff
@@ -0,0 +1,144 @@
+/**
+ * @file optimisticadam_update.hpp
+ * @author Moksh Jain
+ *
+ * OptmisticAdam optimizer. Optimistic Adam is an an algorithm which uses
+ * Optimistic Mirror Descent with the Adam optimizer.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_ADAM_OPTIMISTICADAM_UPDATE_HPP
+#define MLPACK_CORE_OPTIMIZERS_ADAM_OPTIMISTICADAM_UPDATE_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace optimization {
+
+/**
+ * OptimisticAdam is an optimizer which uses Optmistic Mirror Descent with
+ * the Adam Optimizer.
+ *
+ * For more information, see the following.
+ *
+ * @code
+ * @article{
+ *   author  = {Constantinos Daskalakis, Andrew Ilyas, Vasilis Syrgkanis, 
+ *              Haoyang Zeng},
+ *   title   = {Training GANs with Optimism},
+ *   year    = {2017},
+ *   url     = {https://openreview.net/forum?id=SJJySbbAZ}
```

Comment:
Just a suggestion, you could cite the `arXiv` link here (https://arxiv.org/abs/1711.00141).

Reply:
Sure, I'll do that. Any specific reason why the arxiv link over this one?

Reply:
I find it useful to access the different concrete revisions there, instead of multiple revisions that I saw on `openreview`. But again, it's just a suggestion.